llvm/tools/sycl-post-link/ModuleSplitter.cpp

```diff
@@ -891,5 +930,42 @@ getDeviceCodeSplitter(ModuleDesc &&MD, IRSplitMode Mode, bool IROutputOnly,
   return std::make_unique<ModuleCopier>(std::move(MD), std::move(Groups));
 }
 
+SmallVector<ModuleDesc, 2> splitByESIMD(ModuleDesc &&MD,
+                                        bool EmitOnlyKernelsAsEntryPoints) {
+
+  SmallVector<module_split::ModuleDesc, 2> Result;
+
+  EntryPointGroupVec EntryPointGroups =
+      groupEntryPointsByKernelType(MD, EmitOnlyKernelsAsEntryPoints);
```

Comment:
I see that by kernel 'type', you mean whether it's ESIMD or not. I thought this name was a bit ambiguous. Can we pls rename groupEntryPointsByESIMDType?

Again, only a 'nit'. Pls feel free to ignore.

Thanks

Reply:
`groupEntryPointsByKernelType` is an existing function and not something I've introduced. I agree that it could be named better, but I would prefer to do renaming in a separate PR to avoid overloading this one with unrelated changes